### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Ruslan-Mykhalchak**
[DEMO LINK](https://Ruslan-Mykhalchak.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by Saharij — APPROVED** [review]
Well done!!!

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +15,127 @@ const goodsFromServer: string[] = [
...
+  isVisible: boolean,
+  goods: string[],
+  isReversed: boolean,
+  sortBy: 'alphabet' | 'length' | null,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isVisible: false,
+    goods: goodsFromServer,
```
Comment (aliceblunt8) [inline src/App.tsx:28]: add copy to the state

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,127 @@ const goodsFromServer: string[] = [
...
+              type="button"
+              onClick={this.reset}
+            >
+              Reset
+            </button>
+            <ul className="list">
+              {visibleGoods.map(item => (
+                <li
+                  key={item}
+                >
```
Comment (aliceblunt8) [inline src/App.tsx:128]:
```suggestion
                <li key={item}>
```

**Review by aliceblunt8 — APPROVED** [review]
GJ

**Pull request closed without merging**